src/Cryptol/IR/FreeVars.hs

```diff
@@ -118,7 +118,7 @@ instance FreeVars Expr where
       EAbs x t e        -> freeVars t <> rmVal x (freeVars e)
       EProofAbs p e     -> freeVars p <> freeVars e
       EProofApp e       -> freeVars e
-      EWhere e ds       -> freeVars ds <> rmVals (defs ds) (freeVars e)
+      EWhere e ds       -> rmVals (defs ds) (freeVars ds) <> rmVals (defs ds) (freeVars e)
```

Comment:
This line was certainly wrong before, but I don't think the new version is quite right either, as it doesn't reflect the scoping rules. When an `EWhere` has multiple `DeclGroup`s, variables defined by a `DeclGroup` are only in scope in the `DeclGroup`s that come *after* it in the list.

To do this properly, I think we'll need to define a helper function (similar to the one used for the `EComp` case) that computes the free variable set by doing a fold over the list of `DeclGroup`s.